BrofilerApp/ViewModels/PlatformSelectorViewModel.cs

```diff
@@ -42,15 +119,224 @@ public PlatformSelectorViewModel()
+
+            await Task.Run(() =>
+            {
+                try
+                {
+                    var connection = _config.LocalSettings.Data.Connections.FirstOrDefault(x => x.Address == platform.IP && x.Port == platform.Port);
+
+                    if (connection == null)
+                        connection = _config.LocalSettings.Data.Connections.FirstOrDefault(x => x.Name == platform.Name);
+
+                    bool count = _config.LocalSettings.Data.Connections.Remove(connection);
+
+                    if (count)
+                        _config.LocalSettings.Data.Connections.Add(new Platform.Connection()
+                        { Name = platform.Name, Address = platform.IP, Port = platform.Port, Target = platform.PlatformType });
 
-            ActivePlatform = Platforms[0];
+                    _config.LocalSettings.Save();
+                }
+                catch (Exception)
+                {
+                }
+
+            });
+        }
+
+        // Save all custom collection to LocalSettings
+        private async Task SaveAllCustomConnectionsToLocalSettingsAsync()
+        {
+            var customConnectios = Platforms.Where(x => x.CoreType == false);
+
+            _isSelfUpdateFlag = true;
+
+            await Task.Run(() =>
+            {
+                try
+                {
+                    if (_config.LocalSettings.Data.Connections != null)
+                        _config.LocalSettings.Data.Connections.Clear();
+                    else
+                        _config.LocalSettings.Data.Connections = new List<Platform.Connection>();
+
+                    foreach (var item in customConnectios)
+                        _config.LocalSettings.Data.Connections.Add(new Platform.Connection() { Name = item.Name, Address = item.IP, Port = item.Port, Target = item.PlatformType });
+
+                    _config.LocalSettings.Save();
+                }
+                catch (Exception)
+                {
+                }             
+            } );
+        }
+
+        // Update collection platforms if LocalSettings changed by other application
+        private void LocalSettings_OnChanged()
+        {
+            if  (!_isSelfUpdateFlag)
+                 Application.Current.Dispatcher.BeginInvoke(new Action(() =>SetCustomPlatformsFromLocalSettings() ));
+
+            _isSelfUpdateFlag = false;
+        }
+
+        private bool IsIpValid(string ip)
+        {
+            if (ip?.Length > 0)
+            {
+                IPAddress address = null;
+                return IPAddress.TryParse(ip, out address);
+            }
+            else
+                return false;
+        }
+
+        private async Task AddPlatformAsync(IPAddress ip, short port, bool autofocus)
+        {
+            if (ip.Equals(IPAddress.None) || ip.Equals(IPAddress.Any) || ip.Equals(IPAddress.Loopback))
+                return;
+
+            PingReply reply = await new Ping().SendPingAsync(ip, 16);
```

Comment:
Too small timeout. Moreover some of the network architectures block ping request.
We can't really rely on ping in this case.